pkg/auth/mutual_authhandler.go

```diff
@@ -140,35 +141,35 @@ func (m *mtlsAuthHandler) authenticate(ar *authRequest) (*authResponse, error) {
 	}, nil
 }
 
-func (m *mtlsAuthHandler) authType() policy.AuthType {
-	return policy.AuthTypeMTLSSpiffe
+func (m *mutualAuthHandler) authType() policy.AuthType {
+	return policy.AuthTypeAuthSpire
```

Comment:
having the SPIRE/SPIFFE agnostic "mutual authhandler" returning type "spire" is ok for the time being. we should refactor this once we provide support for multiple CertificateProviders.

Reply:
yeah, we might need to revisit this one later :(